Integrations/BluecatAddressManager/BluecatAddressManager.py

```diff
@@ -0,0 +1,393 @@
+        :param headers: Request headers
+
+        :type safe: ``bool``
+        :param safe: If set to true will return None in case of http error
+
+        :return: Returns the http request response json
+        :rtype: ``dict``
+    """
+    headers['Authorization'] = get_token()
+    url = BASE_URL + url_suffix
+    try:
+        res = requests.request(method, url, verify=USE_SSL, params=params, data=data, headers=headers)
+        # Try to create a new token
+        if res.status_code == 401:
+            headers['Authorization'] = get_token(new_token=True)
+            res = requests.request(method, url, verify=USE_SSL, params=params, data=data, headers=headers)
+    except requests.exceptions.RequestException:
+        return_error('Error in connection to the server. Please make sure you entered the URL correctly.')
+    # Handle error responses gracefully
+    if res.status_code not in {200, 201, 202}:
+        try:
+            result_msg = res.json()
+        finally:
+            reason = result_msg if result_msg else res.reason
+            err_msg = f'Error in API call. code:{res.status_code}; reason: {reason}'
+            if safe:
+                return None
+            return_error(err_msg)
+    return res.json()
+
+
+def get_token(new_token=False):
+    """
+        Retrieves the token from the server if it's expired and updates the global HEADERS to include it
+
+        :param new_token: If set to True will generate a new token regardless of time passed
+
+        :rtype: ``str``
+        :return: Token
+    """
+    now = datetime.now()
+    ctx = demisto.getIntegrationContext()
+    if ctx and not new_token:
+        passed_mins = get_passed_mins(now, ctx.get('time'))
+        if passed_mins >= TOKEN_LIFE_TIME_MINS:
+            # token expired
+            auth_token = get_token_request()
+            demisto.setIntegrationContext({'auth_token': auth_token, 'time': date_to_timestamp(now) / 1000})
+        else:
+            # token hasn't expired
+            auth_token = ctx.get('auth_token')
+    else:
+        # generating new token
+        auth_token = get_token_request()
+        demisto.setIntegrationContext({'auth_token': auth_token, 'time': date_to_timestamp(now) / 1000})
+    return auth_token
+
+
+def get_configuration():
+    """
+    Gets the chosen configuration to run queries on
+
+    :return: User configuration id, or the first configuration id if no user configuration provided
+    """
+    user_conf = USER_CONF
+    params = {
+        'type': 'Configuration',
+        'start': 0,
+        'count': 100
+    }
+    confs = http_request('GET', '/getEntities', params)
+    if not confs:
+        return_error('No configurations could be fetched from the system')
+    if user_conf:
+        for conf in confs:
+            if conf.get('name') == user_conf:
+                return conf.get('id')
+
+    return confs[0].get('id')
+
+
+def get_passed_mins(start_time, end_time_str):
```

Comment:
I think it is better that this function will get end time in datetime. using 2 different input types for the same information is strange.